internal/testutils/testmain/windows.go

```diff
@@ -0,0 +1,219 @@
+package testmain
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+)
+
+type tracelogKeywords uint64
+
+// Know tracelog keywords.
+//
+// See https://github.com/microsoft/ebpf-for-windows/blob/main/libs/shared/ebpf_tracelog.h
+var allKeywords = []string{
+	"entry-exit",
+	"base",
+	"error",
+	"epoch",
+	"core",
+	"link",
+	"map",
+	"program",
+	"api",
+	"printk",
+	"native",
+}
+
+func (kw *tracelogKeywords) UnmarshalText(text []byte) error {
+	decoded, err := strconv.ParseUint(string(text), 0, 64)
+	if err != nil {
+		return fmt.Errorf("foo: %w", err)
+	}
+	*kw = tracelogKeywords(decoded)
+	return nil
+}
+
+func (kw tracelogKeywords) decode() []string {
+	var keywords []string
+	for _, keyword := range allKeywords {
+		if kw&1 > 0 {
+			keywords = append(keywords, keyword)
+		}
+		kw >>= 1
+	}
+	if kw > 0 {
+		keywords = append(keywords, fmt.Sprintf("0x%x", kw))
+	}
+	return keywords
+}
+
+type traceSession struct {
+	session string
+}
+
+// newTraceSession starts a trace log for eBPF for Windows related events.
+//
+// * https://github.com/microsoft/ebpf-for-windows/blob/main/docs/GettingStarted.md#using-tracing
+// * https://devblogs.microsoft.com/performance-diagnostics/controlling-the-event-session-name-with-the-instance-name/ and
+func newTraceSession() (*traceSession, error) {
+	def := filepath.Join(os.Getenv("ProgramFiles"), "ebpf-for-windows\\ebpfforwindows.wprp")
+	if _, err := os.Stat(def); err != nil {
+		return nil, err
+	}
+
+	session := fmt.Sprintf("epbf-go-%d", os.Getpid())
+	wpr := exec.Command("wpr.exe", "-start", def, "-filemode", "-instancename", session)
+	wpr.Stderr = os.Stderr
+	if err := wpr.Run(); err != nil {
+		return nil, err
+	}
+
+	return &traceSession{session}, nil
+}
+
+func (ts *traceSession) Close() error {
+	if ts == nil {
+		return nil
+	}
+
+	return ts.stop(os.DevNull)
+}
+
+func (ts *traceSession) stop(file string) error {
+	if ts.session == "" {
+		return nil
+	}
+
+	wpr := exec.Command("wpr.exe", "-stop", file, "-instancename", ts.session)
```

Comment:
Is the wpr.exe executable available by default on windows systems? There is at least one other shell-out in this PR, the same question applies there.

Reply:
Good question, I think `wpr.exe` does ship with Windows:
```
PS C:\Users\lorenz> (get-command wpr.exe).Path
C:\Windows\system32\wpr.exe
```

Reply:
wpr.exe does come out of the box: https://learn.microsoft.com/en-us/windows-hardware/test/wpt/introduction-to-wpr#wpr-command-line-and-user-interface

> WPR.exe ships with Windows OS (Windows 8.1 or later) and you do not need additional installation.

Netsh seems to be the `ip` equivalent: https://learn.microsoft.com/en-us/windows-server/networking/technologies/netsh/netsh

```
PS C:\Users\lorenz> (get-command netsh.exe).Path
C:\Windows\system32\netsh.exe
```